### DISCUSSION TITLE
about vuetify

### DISCUSSION THREAD

**Discussion by ZzZuf** [Kitware/trame#499] [Q&A] [answered]
Dear trame and vuetify experts,
I am new at trame and vuetify and I have a question about Vuetify in Trame.
If I want to implement such a feature in a web page, what kind of components should I choose
![1fa956f123ef1d18b4b3844cacf6fbb](https://github.com/Kitware/trame/assets/162105830/19ee5b17-d87a-4b6a-931b-2df84c1bd741)

**Comment by jourdain** [comment]
First you should figure out your vuetify code for your UI. And for that you can use [their playground](https://play.vuetifyjs.com/).
Otherwise, the trame community can't really help for side project that we leverage (vue, vuetify, plotly, vtk, paraview...). What we can do, is help you transpose that knowledge in the context of trame.

But if you are looking for more in depth help, don't hesitate to reach out to [Kitware for support](https://www.kitware.com/contact/advanced-support/).

**Reply by ZzZuf** [reply]
with vuetify.VRow(classes="pt-2", dense=True):
                with vuetify.VCheckbox(v_model=("customCircle", False), label="customCircle"):
                    with vuetify.VCol(cols=6):
                        vuetify.VTextField(
                            label="Radius",
                            v_model=('circleRadius', AlgorithmParam.gethsapce()),
                            disabled="!customCircle",
                            hint="Enter radius",
                            outlined=True,
                            clearable=True,
                        )
            with vuetify.VRow(classes="pt-2", dense=True):
                with vuetify.VCheckbox(v_model=("customRect", False), label="Custom Rectangle"):
                    with vuetify.VCol(cols=6):
                        vuetify.VTextField(
                            label="Length",
                            v_model=('rectLength', AlgorithmParam.gethsapce()),
                            disabled="!customRect",
                            hint="Enter length",
                            outlined=True,
                            clearable=True,
                        )
                    with vuetify.VCol(cols=6):
                        vuetify.VTextField(
                            label="width",
                            v_model=('rectWidth', AlgorithmParam.gethsapce()),
                            disabled="!customRect",
                            hint="Enter width",
                            outlined=True,
                            clearable=True,
                        )
                    with vuetify.VCol(cols=6):
                        vuetify.VTextField(
                            label="height",
                            v_model=('rectHeight', AlgorithmParam.gethsapce()),
                            disabled="!customRect",
                            hint="Enter height",
                            outlined=True,
                            clearable=True,
                        )
Here's my code about VCheckbox, but its VTextFiled doesn't show up。
![image](https://github.com/Kitware/trame/assets/162105830/f3e81b64-3b4f-483f-bf82-e82631b3d058)

**Reply by jourdain** [reply]
You can't nest things under the checkbox. Those will be skipped since the checkbox don't expect any slot.

**Reply by jourdain** [reply]
```
with row:
  with col:
      checkbox
  with col
      textfield

with row:
  with col:
      checkbox
  with col
      textfield
```

**Reply by ZzZuf** [reply]
Now there's a new issue where the text field is forbidden regardless of whether I check the box or not.

![image](https://github.com/Kitware/trame/assets/162105830/891a05cb-ffa1-43b3-a82a-7db69ff5a46b)
 with vuetify.VRow(classes="pt-1", dense=True):
                html.Span("挖孔", style="color: black; font-weight: bold;")
            with vuetify.VRow(classes="pt-2", dense=True):
                with vuetify.VCol(cols=6):
                    vuetify.VCheckbox(v_model=("customCircle", True), label="自定义圆")
                with vuetify.VCol(cols=6):
                    vuetify.VTextField(
                        label="Radius",
                        v_model=('circleRadius', AlgorithmParam.gethsapce()),
                        disabled= '!customCircle',
                        hint="Enter radius",
                        outlined=True,
                        clearable=True,
                    )
            with vuetify.VRow(classes="pt-2", dense=True):
                with vuetify.VCol(cols=6):
                    vuetify.VCheckbox(v_model=("customRect", False), label="Custom Rectangle")
            with vuetify.VRow(classes="pt-2", dense=True):
                with vuetify.VCol(cols=4):
                    vuetify.VTextField(
                        label="Length",
                        v_model=('rectLength', AlgorithmParam.gethsapce()),
                        disabled= '!customRect' ,
                        hint="Enter length",
                        outlined=True,
                        clearable=True,
                    )
                with vuetify.VCol(cols=4):
                    vuetify.VTextField(
                        label="width",
                        v_model=('rectWidth', AlgorithmParam.gethsapce()),
                        disabled="!customRect",
                        hint="Enter width",
                        outlined=True,
                        clearable=True,
                    )
                with vuetify.VCol(cols=4):
                    vuetify.VTextField(
                        label="height",
                        v_model=('rectHeight', AlgorithmParam.gethsapce()),
                        disabled="!customRect",
                        hint="Enter height",
                        outlined=True,
                        clearable=True,
                    )

**Reply by jourdain** [reply, marked as answer]
Please read the documentation. If you don't want the property `disabled` to be given a string but the value of an expression, you need to provide a tuple. Otherwise, we can not differentiate it from a string.

```python
vuetify.VTextField(
   label="Length",
   v_model=('rectLength', AlgorithmParam.gethsapce()),
   disabled=('!customRect',) ,
   hint="Enter length",
   outlined=True,
   clearable=True,
)
```
